src/WeightList.php

```diff
@@ -24,15 +24,17 @@ public function set($value, $weight)
 
     private function getValueByPosition($position)
     {
+        if(count($this->list) === 0) {
+            throw new \OutOfBoundsException('Index of ' . $position . ' is out of bounds for the array.');
+        }
+
         $accumulator = 0;
         foreach($this->list as $value => $weight) {
             $accumulator += $weight;
             if($position <= $accumulator) {
                 return $value;
             }
         }
-
-        return $value;
```

Comment:
It's a bug. Code doesn't return result if position more than accumulator, it must return last value in this case